src/Skyflow.ts

```diff
@@ -101,81 +117,115 @@ class Skyflow {
       },
       this.#metadata,
     );
-    this.#pureJsController = new PureJsController(this.#client);
-    bus.target(properties.IFRAME_SECURE_ORGIN)
-      .on(ELEMENT_EVENTS_TO_IFRAME.GET_BEARER_TOKEN,
-        (data, callback) => {
-          if (this.#client.config.getBearerToken
-        && (!this.#bearerToken || !isTokenValid(this.#bearerToken))
-          ) {
-            this.#client.config.getBearerToken().then((bearerToken) => {
+    this.#logLevel = config?.options?.logLevel || LogLevel.PROD;
+    this.#pureJsController = new PureJsController(this.#client, { logLevel: this.#logLevel });
+
+    const { showInfoLogs, showErrorLogs } = LogLevelOptions[this.#logLevel];
+
+    this.#showInfoLogs = showInfoLogs;
+    this.#showErrorLogs = showErrorLogs;
+
+    bus
+      .target(properties.IFRAME_SECURE_ORGIN)
+      .on(ELEMENT_EVENTS_TO_IFRAME.GET_BEARER_TOKEN, (data, callback) => {
+        printLog(logs.infoLogs.CAPTURED_BEARER_TOKEN_EVENT, MessageType.INFO,
+          this.#showErrorLogs, this.#showInfoLogs);
+        if (
+          this.#client.config.getBearerToken
+          && (!this.#bearerToken || !isTokenValid(this.#bearerToken))
+        ) {
+          this.#client.config
+            .getBearerToken()
+            .then((bearerToken) => {
+              printLog(logs.infoLogs.BEARER_TOKEN_RESOLVED, MessageType.INFO,
+                this.#showErrorLogs, this.#showInfoLogs);
               this.#bearerToken = bearerToken;
               callback({ authToken: this.#bearerToken });
-            }).catch((err) => {
+            })
+            .catch((err) => {
+              printLog(logs.errorLogs.BEARER_TOKEN_REJECTED, MessageType.ERROR,
+                this.#showErrorLogs, this.#showInfoLogs);
               callback({ error: err });
             });
-          } else {
-            callback({ authToken: this.#bearerToken });
-          }
-        });
+        } else {
+          // ask
+          callback({ authToken: this.#bearerToken });
+        }
+      });
+    printLog(logs.infoLogs.BEARER_TOKEN_LISTENER, MessageType.INFO,
+      this.#showErrorLogs, this.#showInfoLogs);
   }
 
   static init(config: ISkyflow): Skyflow {
+    const {
+      showInfoLogs,
+      showErrorLogs,
+    } = LogLevelOptions[config?.options?.logLevel || LogLevel.PROD];
+    printLog(logs.infoLogs.INITIALIZE_CLIENT, MessageType.INFO, showErrorLogs, showInfoLogs);
     if (
       !config
       || !config.vaultID
       || !isValidURL(config.vaultURL)
       || !config.getBearerToken
     ) {
-      throw new Error('Invalid client credentials');
+      throw new Error(logs.errorLogs.INVALID_CREDENTIALS);
     }
-
     const tempConfig = config;
     tempConfig.vaultURL = config.vaultURL.slice(-1) === '/'
       ? config.vaultURL.slice(0, -1)
       : config.vaultURL;
-
+    printLog(logs.infoLogs.CLIENT_INITIALIZED, MessageType.INFO, showErrorLogs, showInfoLogs);
```

Comment:
this log should be after intilalization. i,e
const skyflow = new Skyflow(config)
printLog(...)
return skyflow